File: ios/SpentyAI/SpentyAI/Features/Billing/LifetimeOfferSheet.swift
import SwiftUI

// MARK: - Lifetime Offer Sheet
// Two modes:
//   showTimer = true  → non-subscriber monthly intercept; 30-minute countdown displayed
//   showTimer = false → existing subscriber upgrade; no timer, always available

struct LifetimeOfferSheet: View {

    // MARK: - Params
    var showTimer: Bool = true
    /// The StoreKit-localised price string for the lifetime_offer product.
    /// Displayed in the header and CTA button so the user sees the exact charge
    /// before confirming (required by App Store guideline 3.1.1).
    var offerPrice: String? = nil
    /// The StoreKit-localised price string for the regular `com.spentyai.lifetime`
    /// product (the price the user will pay AFTER the 30-min offer expires).
    /// Shown next to the discounted offer price so the urgency is verifiable
    /// rather than implied — closes a dark-pattern concern with the timer.
    var fullPrice: String? = nil
    let onAccept: () async -> Void
    let onDecline: () -> Void

    // MARK: - State
    @State private var isPurchasing = false
    @State private var displayTime: String = ""

    // MARK: - Timer
    private let clock = Timer.publish(every: 1, on: .main, in: .common).autoconnect()

    // MARK: - Palette
    private let accentGold = Color(red: 0.831, green: 0.686, blue: 0.216)
    private let darkBg     = Color(red: 0.055, green: 0.122, blue: 0.071)

    // MARK: - Body
    var body: some View {
        VStack(spacing: 0) {
            premiumHeader
            offerContent
        }
        .presentationDetents([.large])
        .presentationDragIndicator(.visible)
        .interactiveDismissDisabled(isPurchasing)
        .onAppear {
            if showTimer {
                LifetimeOfferManager.shared.recordFirstShown()
                displayTime = LifetimeOfferManager.shared.formattedTimeRemaining()
            }
        }
        .onReceive(clock) { _ in
            guard showTimer, !isPurchasing else { return }
            let remaining = LifetimeOfferManager.shared.timeRemaining
            if remaining <= 0 {
                onDecline()
            } else {
                displayTime = LifetimeOfferManager.shared.formattedTimeRemaining()
            }
        }
    }

    // MARK: - Dark premium header
    private var premiumHeader: some View {
        VStack(spacing: 18) {
            Spacer().frame(height: 8)

            // Badge
            HStack(spacing: 5) {
                Image(systemName: showTimer ? "bolt.fill" : "star.fill")
                    .font(.system(size: 10, weight: .bold))
                Text(showTimer ? "ONE-TIME SPECIAL OFFER" : "SUBSCRIBER EXCLUSIVE")
                    .font(.system(size: 11, weight: .bold))
                    .tracking(1.1)
            }
            .foregroundStyle(accentGold)
            .padding(.horizontal, 14)
            .padding(.vertical, 7)
            .background(accentGold.opacity(0.14))
            .clipShape(Capsule())
            .overlay(Capsule().stroke(accentGold.opacity(0.28), lineWidth: 1))

            // Title
            VStack(spacing: 6) {
                Text("Lifetime Access")
                    .font(.system(size: 34, weight: .bold))
                    .foregroundStyle(.white)
                Text("Pay once. Use SpentyAI forever.")
                    .font(.subheadline)
                    .foregroundStyle(.white.opacity(0.6))
            }

            // Price display — show the discounted offer price PROMINENTLY
            // and the regular post-offer price struck-through next to it
            // so the 50%-off claim is verifiable, not just implied. Without
            // this comparison the 30-min countdown looked like a manipulative
            // dark pattern (price never visibly changed during the timer).
            VStack(spacing: 8) {
                Text("One-time purchase")
                    .font(.title3.weight(.medium))
                    .foregroundStyle(.white.opacity(0.38))

                if let price = offerPrice {
                    HStack(alignment: .firstTextBaseline, spacing: 12) {
                        Text(price)
                            .font(.system(size: 36, weight: .bold))
                            .foregroundStyle(.white)
                        if showTimer, let regular = fullPrice, regular != price {
                            Text(regular)
                                .font(.system(size: 20, weight: .semibold))
                                .foregroundStyle(.white.opacity(0.45))
                                .strikethrough(true, color: .white.opacity(0.6))
                        }
                    }
                } else {
                    ProgressView()
                        .tint(.white)
                        .scaleEffect(1.2)
                        .frame(height: 44)
                }

                Text(showTimer
                     ? "LIMITED TIME OFFER  ·  50% OFF"
                     : "EXCLUSIVE LIFETIME RATE")
                    .font(.system(size: 11, weight: .bold))
                    .tracking(0.4)
                    .foregroundStyle(darkBg)
                    .padding(.horizontal, 16)
                    .padding(.vertical, 7)
                    .background(accentGold)
                    .clipShape(Capsule())
            }

            Spacer().frame(height: 4)
        }
        .frame(maxWidth: .infinity)
        .padding(.horizontal, 24)
        .padding(.bottom, 28)
        .background(darkBg)
    }

    // MARK: - Light content section
    private var offerContent: some View {
        VStack(spacing: 0) {
            // Feature rows
            VStack(spacing: 0) {
                offerRow(icon: "infinity",               title: "Use forever, no renewals",     sub: "No annual fees, ever")
                Divider().padding(.leading, 60).padding(.trailing, 20)
                offerRow(icon: "arrow.up.circle.fill",   title: "All future updates included",  sub: "Everything we build, free")
                Divider().padding(.leading, 60).padding(.trailing, 20)
                offerRow(icon: "brain.head.profile",     title: "Full AI features",             sub: "Unlimited insights & analysis")
                Divider().padding(.leading, 60).padding(.trailing, 20)
                offerRow(icon: "shield.lefthalf.filled", title: "Priority support",             sub: "We're here when you need us")
            }
            .padding(.top, 20)

            // Urgency / info note
            urgencyNote
                .padding(.horizontal, 20)
                .padding(.top, 20)

            // Buttons
            VStack(spacing: 14) {
                // Apple App Review 2.1(b): disable CTA until the lifetime_offer
                // price has loaded from StoreKit. Reviewer must not be able to
                // tap a button that triggers an unfulfillable purchase.
                Button {
                    Task {
                        isPurchasing = true
                        await onAccept()
                        isPurchasing = false
                    }
                } label: {
                    Group {
                        if isPurchasing {
                            ProgressView()
                                .tint(.white)
                                .frame(maxWidth: .infinity)
                                .padding(.vertical, 16)
                        } else {
                            HStack(spacing: 0) {
                                VStack(alignment: .leading, spacing: 2) {
                                    Text(offerPrice == nil
                                         ? "Connecting to App Store…"
                                         : (showTimer ? "Get Lifetime Access" : "Upgrade to Lifetime"))
                                        .font(.headline)
                                    if offerPrice != nil {
                                        Text("One-time · no subscriptions")
                                            .font(.caption)
                                            .opacity(0.8)
                                    }
                                }
                                Spacer()
                                // Show actual price in button (guideline 3.1.1)
                                if let price = offerPrice {
                                    Text(price)
                                        .font(.headline)
                                } else {
                                    ProgressView()
                                        .tint(.white)
                                        .scaleEffect(0.8)
                                }
                            }
                            .padding(.horizontal, 20)
                            .padding(.vertical, 16)
                        }
                    }
                    .foregroundStyle(.white)
                    .frame(maxWidth: .infinity)
                    .background(Color.spentyPrimary.opacity(offerPrice == nil ? 0.6 : 1.0),
                                in: RoundedRectangle(cornerRadius: 14))
                }
                .disabled(isPurchasing || offerPrice == nil)

                // Secondary "decline" CTA — needs to read as a real
                // button so users have a visible exit. Previous version
                // was tiny grey text below the giant green primary CTA;
                // users couldn't find it and felt trapped (reported by
                // user 2026-05-10). Now full-width with a visible
                // outline + body-sized text matching the primary.
                Button {
                    onDecline()
                } label: {
                    Text(showTimer ? "No thanks, continue with my plan" : "Maybe later")
                        .font(.body.weight(.semibold))
                        .foregroundStyle(Color.spentyPrimary)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 14)
                        .background(Color.spentyPrimary.opacity(0.06),
                                    in: RoundedRectangle(cornerRadius: 14))
                        .overlay(
                            RoundedRectangle(cornerRadius: 14)
                                .stroke(Color.spentyPrimary.opacity(0.3), lineWidth: 1.2)
                        )
                }
                .disabled(isPurchasing)

                Text("Secure payment via Apple")
                    .font(.caption2)
                    .foregroundStyle(.tertiary)
            }
            .padding(.horizontal, 20)
            .padding(.top, 20)
            .padding(.bottom, 36)
        }
        .background(Color.spentyBgPrimary)
    }

    // MARK: - Urgency / info note (context-aware)
    @ViewBuilder
    private var urgencyNote: some View {
        if showTimer {
            // Timer countdown — wording explicitly states what changes
            // when the timer hits zero (offer goes away, regular price
            // applies) so the urgency is honest rather than implied.
            HStack(spacing: 10) {
                Image(systemName: "clock.fill")
                    .foregroundStyle(.orange)
                VStack(alignment: .leading, spacing: 2) {
                    if let regular = fullPrice {
                        Text("Offer expires in — then \(regular)")
                            .font(.caption)
                            .foregroundStyle(.secondary)
                    } else {
                        Text("Offer expires in")
                            .font(.caption)
                            .foregroundStyle(.secondary)
                    }
                    Text(displayTime)
                        .font(.system(size: 20, weight: .bold, design: .monospaced))
                        .foregroundStyle(.orange)
                }
                Spacer()
                Text("30 min only")
                    .font(.caption2.weight(.semibold))
                    .foregroundStyle(.orange)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 4)
                    .background(Color.orange.opacity(0.12))
                    .clipShape(Capsule())
            }
            .padding(14)
            .frame(maxWidth: .infinity)
            .background(Color.orange.opacity(0.08))
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color.orange.opacity(0.2), lineWidth: 1))
        } else {
            // Subscriber exclusive note
            HStack(spacing: 8) {
                Image(systemName: "star.fill")
                    .font(.caption)
                    .foregroundStyle(Color(red: 0.831, green: 0.686, blue: 0.216))
                Text("Always available — exclusive rate for SpentyAI subscribers")
                    .font(.caption)
                    .foregroundStyle(.secondary)
            }
            .padding(14)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(Color(red: 0.831, green: 0.686, blue: 0.216).opacity(0.07))
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color(red: 0.831, green: 0.686, blue: 0.216).opacity(0.2), lineWidth: 1))
        }
    }

    // MARK: - Feature row
    private func offerRow(icon: String, title: String, sub: String) -> some View {
        HStack(spacing: 16) {
            ZStack {
                Circle()
                    .fill(Color.spentyPrimary.opacity(0.08))
                    .frame(width: 36, height: 36)
                Image(systemName: icon)
                    .font(.system(size: 15))
                    .foregroundStyle(Color.spentyPrimary)
            }
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.subheadline.weight(.semibold))
                    .foregroundStyle(.primary)
                Text(sub)
                    .font(.caption)
                    .foregroundStyle(.secondary)
            }
            Spacer()
            Image(systemName: "checkmark.circle.fill")
                .font(.body)
                .foregroundStyle(Color.spentyPrimary)
        }
        .padding(.horizontal, 20)
        .padding(.vertical, 14)
    }
}

#Preview {
    LifetimeOfferSheet(showTimer: true, offerPrice: "₹4,999", onAccept: {}, onDecline: {})
}

File: android-native/app/src/main/java/com/spentyai/app/features/billing/LifetimeOfferSheet.kt
package com.spentyai.app.features.billing

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AllInclusive
import androidx.compose.material.icons.filled.Bolt
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Psychology
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.Shield
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.Upgrade
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.foundation.clickable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.spentyai.app.core.theme.SpentyPrimary
import com.spentyai.app.core.theme.SpentyType
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

/**
 * Lifetime Offer Sheet — Android port of iOS LifetimeOfferSheet.swift.
 *
 * Two modes:
 *   showTimer = true  → non-subscriber Monthly intercept; 1-hour countdown.
 *   showTimer = false → existing subscriber upgrade; no timer, always available.
 *
 * UI is matched visually to iOS: dark forest-green header with gold accent,
 * "ONE-TIME SPECIAL OFFER" / "SUBSCRIBER EXCLUSIVE" badge, big price,
 * "LIMITED TIME OFFER · 50% OFF" pill, 4 feature rows, urgency note that
 * flips between orange countdown and gold "always available", primary CTA
 * with price right-aligned, secondary text button, "Secure payment via Google
 * Play" footer (only platform-specific copy that differs from iOS).
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LifetimeOfferSheet(
    showTimer: Boolean,
    offerPrice: String?,
    isPurchasing: Boolean = false,
    onAccept: () -> Unit,
    onDecline: () -> Unit
) {
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
    val scope = rememberCoroutineScope()
    val context = LocalContext.current
    val manager = remember { LifetimeOfferManager.shared(context) }

    var displayTime by remember { mutableStateOf(if (showTimer) manager.formattedTimeRemaining() else "") }

    // Start the clock the first time we open the sheet (timer mode only).
    LaunchedEffect(showTimer) {
        if (showTimer) {
            manager.recordFirstShown()
            displayTime = manager.formattedTimeRemaining()
        }
    }

    fun close(after: () -> Unit) {
        scope.launch {
            sheetState.hide()
            after()
        }
    }

    // 1-second tick. Auto-decline when the window expires. Routed through the
    // same close path as the user-tap "No thanks" so the dismiss animation
    // plays in both cases (review nit #4).
    LaunchedEffect(showTimer, isPurchasing) {
        if (!showTimer || isPurchasing) return@LaunchedEffect
        while (true) {
            delay(1000L)
            if (manager.timeRemainingMs <= 0L) {
                close { onDecline() }
                break
            }
            displayTime = manager.formattedTimeRemaining()
        }
    }

    ModalBottomSheet(
        onDismissRequest = {
            // Mirror iOS interactiveDismissDisabled(isPurchasing) — block swipe-down
            // mid-purchase, otherwise treat dismiss as decline.
            if (!isPurchasing) onDecline()
        },
        sheetState = sheetState,
        dragHandle = null,
        containerColor = Color.Transparent
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(MaterialTheme.colorScheme.background)
        ) {
            PremiumHeader(showTimer = showTimer, offerPrice = offerPrice)
            OfferContent(
                showTimer = showTimer,
                offerPrice = offerPrice,
                displayTime = displayTime,
                isPurchasing = isPurchasing,
                onAccept = onAccept,
                onDecline = { close { onDecline() } }
            )
        }
    }
}

// MARK: - Brand palette (mirrors iOS exactly)

/** iOS accentGold = Color(red: 0.831, green: 0.686, blue: 0.216). */
private val AccentGold = Color(0xFFD4AF37)

/** iOS darkBg = Color(red: 0.055, green: 0.122, blue: 0.071). */
private val DarkBg = Color(0xFF0E1F12)

private val OrangeWarning = Color(0xFFFF9500)

@Composable
private fun PremiumHeader(showTimer: Boolean, offerPrice: String?) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(DarkBg)
            .padding(horizontal = 24.dp)
            .padding(top = 16.dp, bottom = 28.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(18.dp)
    ) {
        // Badge
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(5.dp),
            modifier = Modifier
                .clip(CircleShape)
                .background(AccentGold.copy(alpha = 0.14f))
                .border(1.dp, AccentGold.copy(alpha = 0.28f), CircleShape)
                .padding(horizontal = 14.dp, vertical = 7.dp)
        ) {
            Icon(
                imageVector = if (showTimer) Icons.Default.Bolt else Icons.Default.Star,
                contentDescription = null,
                tint = AccentGold,
                modifier = Modifier.size(12.dp)
            )
            Text(
                text = if (showTimer) "ONE-TIME SPECIAL OFFER" else "SUBSCRIBER EXCLUSIVE",
                style = SpentyType.Caption2.copy(
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 1.1.sp
                ),
                color = AccentGold
            )
        }

        // Title block
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(6.dp)
        ) {
            Text(
                text = "Lifetime Access",
                style = SpentyType.Title1.copy(fontSize = 34.sp, fontWeight = FontWeight.Bold),
                color = Color.White
            )
            Text(
                text = "Pay once. Use SpentyAI forever.",
                style = SpentyType.Subheadline,
                color = Color.White.copy(alpha = 0.6f)
            )
        }

        // Price block
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Text(
                text = "One-time purchase",
                style = SpentyType.Title3.copy(fontWeight = FontWeight.Medium),
                color = Color.White.copy(alpha = 0.38f)
            )
            if (offerPrice != null) {
                Text(
                    text = offerPrice,
                    style = SpentyType.Title1.copy(fontSize = 36.sp, fontWeight = FontWeight.Bold),
                    color = Color.White
                )
            } else {
                CircularProgressIndicator(
                    modifier = Modifier
                        .height(44.dp)
                        .size(28.dp),
                    color = Color.White,
                    strokeWidth = 2.5.dp
                )
            }
            Text(
                text = "LIMITED TIME OFFER  ·  50% OFF",
                style = SpentyType.Caption2.copy(
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 0.4.sp
                ),
                color = DarkBg,
                modifier = Modifier
                    .clip(CircleShape)
                    .background(AccentGold)
                    .padding(horizontal = 16.dp, vertical = 7.dp)
            )
        }
    }
}

@Composable
private fun OfferContent(
    showTimer: Boolean,
    offerPrice: String?,
    displayTime: String,
    isPurchasing: Boolean,
    onAccept: () -> Unit,
    onDecline: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(MaterialTheme.colorScheme.background)
    ) {
        Spacer(modifier = Modifier.height(20.dp))

        OfferRow(
            icon = Icons.Default.AllInclusive,
            title = "Use forever, no renewals",
            sub = "No annual fees, ever"
        )
        FeatureDivider()
        OfferRow(
            icon = Icons.Default.Upgrade,
            title = "All future updates included",
            sub = "Everything we build, free"
        )
        FeatureDivider()
        OfferRow(
            icon = Icons.Default.Psychology,
            title = "Full AI features",
            sub = "Unlimited insights & analysis"
        )
        FeatureDivider()
        OfferRow(
            icon = Icons.Default.Shield,
            title = "Priority support",
            sub = "We're here when you need us"
        )

        Spacer(modifier = Modifier.height(20.dp))

        Box(modifier = Modifier.padding(horizontal = 20.dp)) {
            UrgencyNote(showTimer = showTimer, displayTime = displayTime)
        }

        Spacer(modifier = Modifier.height(20.dp))

        // CTA stack
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Primary CTA
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(14.dp))
                    .background(SpentyPrimary)
                    .then(
                        if (!isPurchasing) Modifier.clickable(onClick = onAccept)
                        else Modifier
                    )
                    .padding(horizontal = 20.dp, vertical = 16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                if (isPurchasing) {
                    Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
                        CircularProgressIndicator(
                            color = Color.White,
                            strokeWidth = 2.5.dp,
                            modifier = Modifier.size(22.dp)
                        )
                    }
                } else {
                    Column(
                        modifier = Modifier.weight(1f),
                        verticalArrangement = Arrangement.spacedBy(2.dp)
                    ) {
                        Text(
                            text = if (showTimer) "Get Lifetime Access" else "Upgrade to Lifetime",
                            style = SpentyType.Headline,
                            color = Color.White
                        )
                        Text(
                            text = "One-time · no subscriptions",
                            style = SpentyType.Caption1.copy(fontSize = 12.sp),
                            color = Color.White.copy(alpha = 0.8f)
                        )
                    }
                    if (offerPrice != null) {
                        Text(
                            text = offerPrice,
                            style = SpentyType.Headline,
                            color = Color.White
                        )
                    } else {
                        CircularProgressIndicator(
                            color = Color.White,
                            strokeWidth = 2.dp,
                            modifier = Modifier.size(18.dp)
                        )
                    }
                }
            }

            // Secondary text button
            TextButton(
                onClick = onDecline,
                enabled = !isPurchasing
            ) {
                Text(
                    text = if (showTimer) "No thanks, go back" else "Maybe later",
                    style = SpentyType.Subheadline,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            // Footer — only platform-specific copy that differs from iOS
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.Lock,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.6f),
                    modifier = Modifier.size(11.dp)
                )
                Text(
                    text = "Secure payment via Google Play",
                    style = SpentyType.Caption2.copy(fontSize = 11.sp),
                    color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.6f),
                    textAlign = TextAlign.Center
                )
            }
        }

        Spacer(modifier = Modifier.height(36.dp))
    }
}

@Composable
private fun UrgencyNote(showTimer: Boolean, displayTime: String) {
    if (showTimer) {
        // Orange countdown
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(12.dp))
                .background(OrangeWarning.copy(alpha = 0.08f))
                .border(1.dp, OrangeWarning.copy(alpha = 0.2f), RoundedCornerShape(12.dp))
                .padding(14.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Schedule,
                contentDescription = null,
                tint = OrangeWarning
            )
            Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                Text(
                    text = "Offer expires in",
                    style = SpentyType.Caption1,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Text(
                    text = displayTime,
                    style = SpentyType.Title3.copy(
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold,
                        fontFamily = FontFamily.Monospace
                    ),
                    color = OrangeWarning
                )
            }
            Text(
                text = "1 hour only",
                style = SpentyType.Caption2.copy(fontSize = 10.sp, fontWeight = FontWeight.SemiBold),
                color = OrangeWarning,
                modifier = Modifier
                    .clip(CircleShape)
                    .background(OrangeWarning.copy(alpha = 0.12f))
                    .padding(horizontal = 8.dp, vertical = 4.dp)
            )
        }
    } else {
        // Gold "always available" note
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(12.dp))
                .background(AccentGold.copy(alpha = 0.07f))
                .border(1.dp, AccentGold.copy(alpha = 0.2f), RoundedCornerShape(12.dp))
                .padding(14.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Star,
                contentDescription = null,
                tint = AccentGold,
                modifier = Modifier.size(14.dp)
            )
            Text(
                text = "Always available — exclusive rate for SpentyAI subscribers",
                style = SpentyType.Caption1,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
    }
}

@Composable
private fun OfferRow(icon: ImageVector, title: String, sub: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Box(
            modifier = Modifier
                .size(36.dp)
                .clip(CircleShape)
                .background(SpentyPrimary.copy(alpha = 0.08f)),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = SpentyPrimary,
                modifier = Modifier.size(18.dp)
            )
        }
        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(title, style = SpentyType.Subheadline.copy(fontWeight = FontWeight.SemiBold))
            Text(
                sub,
                style = SpentyType.Caption1,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
        Icon(
            imageVector = Icons.Default.CheckCircle,
            contentDescription = null,
            tint = SpentyPrimary,
            modifier = Modifier.size(20.dp)
        )
    }
}

@Composable
private fun FeatureDivider() {
    HorizontalDivider(
        modifier = Modifier.padding(start = 60.dp, end = 20.dp),
        thickness = 0.5.dp,
        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.08f)
    )
}

